### ISSUE TITLE
Do we need MultiGPULossCompute?

### ISSUE THREAD

**Issue by alexyalunin** [harvardnlp/annotated-transformer#36]
Since we have 
`model_par = torch.nn.DataParallel(model, devices=devices)`
do we need MultiGPULossCompute? 
I thought pytorch comes with targets scattering and other stuff inside MultiGPULossCompute prebuilt in DataParallel.

I faced with some empty error while using MultiGPULossCompute, so I went with SimpleLossCompute and everything was fine.

**Comment by ghost** [comment]
DataParallel only replicates the model and scatters the input to different GPU. But the output has to been gathered at GPU[0](by default) to calculate the loss and do BP. The MultiGPULoss function implements distributed loss calculation and word generations, making the training process faster.
